kubel.el

```diff
@@ -702,6 +706,36 @@ P can be a single number or a localhost:container port pair."
 						                              (kubel--get-resource-under-cursor)
 						                            (kubel--select-resource "Pods")))))
 
+(defun kubel-exec-shell-pod ()
+  "Exec into the pod under the cursor -> shell."
+  (interactive)
+  (kubel-setup-tramp)
+  (let* ((dir-prefix (or
+                    (when (tramp-tramp-file-p default-directory)
+                      (with-parsed-tramp-file-name default-directory nil
+                        (format "%s%s:%s@%s|" (or hop "") method user host))) ""))
+         (pod (if (kubel--is-pod-view)
+                  (kubel--get-resource-under-cursor)
+                (kubel--select-resource "Pods")))
+         (default-directory (format "/%skubectl:%s:/" dir-prefix pod)))
+    (shell (format "*kubel - shell - %s*" pod))))
+
+(defvar eshell-buffer-name)
```

Comment:
why do we need a `defvar` here? is it to calm flycheck? and should eshell be required?

Reply:
Sorry, I miss your comment. I guess you will ask this question :P 

I learn this from https://github.com/Silex/docker.el/blob/master/docker-container.el#L106, seems eshell require it. I don't know if we have another better way

Reply:
interesting. I will take a look see if it's still necessary.

Reply:
BTW, Do you think should rename buffer format from `kubel - shell -  %s` to `kubel:shell:%s`. I think it is short than (Just my opinion)

Reply:
Totally up to you. I like it shorter too but I have no idea if there are any conventions about it. Feel free to change if you want to.

Reply:
I mean all format buffer name in kubel. It exactlly up to you. 
I don't know if there are any conventions about it either. I just see in TRAMP, it is `ssh|method:host:/path` so I think using `:` to separate components

Reply:
It's true that other kubel buffers use a `-`. Let's keep it as is for now and we can always revisit.